index.ts

```diff
@@ -597,4 +597,17 @@ function tsdd(suite) {
 		context.skipOnError = skipOnError;
 	});
 };
+
+let userContainer:Container;
+
+export interface Container {
+	get(klass: any): any
+}
+export function useContainer(container: Container) {
+	userContainer = container;
+}
+
+const getInstance = (klass:any) => userContainer ? userContainer.get(klass) : new klass();
```

Comment:
Register a default userContainer rather than if-ing here

Reply:
@pana-cc over in junit5 land they allow injection on all non static members of a class, including fields. So we should at least also provide for an abstraction of the `@inject` decorator.

Furthermore, they allow parameter injection on both hooks, say `beforeEach` and `afterEach` and also on the test method level.

I wonder how one could define the injection system so that it can be made to handle a specific test class.

And considering async methods this might require additional investigation into how individual ioc handle this specific case.

Reply:
Absolutely, that's why I'd prefer to start with an abstract DependencyInjectionSystem, where if one wants to use a junit flavored one, we will be able to provide enough API in the interface in future and delegate calls to before/after and the actual test to the DI like:
```
di.call(instance, method, done?, params?);
```
Where the DI will be able to inspect custom decorators etc.

Reply:
@pana-cc 
from you code
```
    // Maybe check if it is not allready added?
    dependencyInjectionSystems.unshift(instantiator);
```
how about use Set<DependencyInjectionSystem> to instead of DependencyInjectionSystem[], in this way we can directly use `Set.has()` to check. Or just use `Array.some()`?